### PR TITLE
Making linter add ^ to version when adding missing deps

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by snowystinger — DISMISSED** [review]

**Comment by devongovett** [comment]
If it's an alpha/beta/rc, it should *not* use a caret range, it should be fixed. This is because we allow breaking changes but technically semver sees later prereleases as compatible.

**Comment by LFDanLu** [comment]
@devongovett Ah, thanks for the explanation, will fix so it detects the alpha/beta/rc when deciding whether or not to add a caret.

File: bin/imports.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const findUp = require('find-up');
 const path = require('path');
 const fs = require('fs');
 const Module = require('module');
+const substrings = ['alpha', 'beta', 'rc'];
```
Comment (snowystinger) [inline bin/imports.js:17]: you can make this more generic
```suggestion
const substrings = ['-', '+'];
```
the major.minor.patch will always be separated from everything else by a dash or a plus https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions

**Review by snowystinger — DISMISSED** [review]

**Comment by dannify** [comment]
Sorry for the conflicts

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]
LGTM

**Pull request merged**